src/OrchardCore.Modules/OrchardCore.Users/Services/SetupEventHandler.cs

```diff
@@ -29,11 +32,12 @@ public SetupEventHandler(IUserService userService)
                 UserName = properties.TryGetValue(SetupConstants.AdminUsername, out var adminUserName) ? adminUserName?.ToString() : String.Empty,
                 UserId = properties.TryGetValue(SetupConstants.AdminUserId, out var adminUserId) ? adminUserId?.ToString() : String.Empty,
                 Email = properties.TryGetValue(SetupConstants.AdminEmail, out var adminEmail) ? adminEmail?.ToString() : String.Empty,
-                RoleNames = new string[] { "Administrator" },
                 EmailConfirmed = true
             };
 
-            return _userService.CreateUserAsync(user, properties[SetupConstants.AdminPassword]?.ToString(), reportError);
+            user.RoleNames.Add(_keyNormalizer.NormalizeName("Administrator"));
```

Comment:
I think if you wan to use `.Add()` you should check if the role exists first or not. Otherwise we could have another handler setting up the same role.

Reply:
I agree, did it too quickly for testing

Reply:
Oh yes, I thought about it but didn't do it as here we create a very `new User()`.

Reply:
Also, what you think about this knowing that now the "Administrator" role may not exist.

For now I think that's okay because when we lookup User RoleNames, as I remember we compare them to existing Roles.

Reply:
@jtkech 
Oh yes I noticed the user is created in the event. Maybe don't use .Add then it would be safer. 

Good point on the roles. I think it is safe to add it since it won't add/remove value unless the roles module is enabled. But, maybe we can set the SuperUser property to true here on the new object here and don't set any role. Then we can add a new implementation to the Users event that would add Administrator rote only when the roles module is enabled. This new event would only be registered when Roles module is enabled

Reply:
Yes I used Add to not realloc a List.

Okay will look at the SuperUser property